src/pages/workspaces/workspace/analysis/modals/AnalysisModal.test.ts

```diff
@@ -0,0 +1,399 @@
+  })
+}))
+
+jest.mock('src/pages/workspaces/workspace/analysis/file-utils', () => {
+  const originalModule = jest.requireActual('src/pages/workspaces/workspace/analysis/file-utils')
+  return {
+    ...originalModule,
+    // getExtension: jest.fn().mockImplementation((ext: Extension) => originalModule.getExtension(ext)), //() => 'ipynb' as Extension,
+    useAnalysisFiles: jest.fn()
+  }
+})
+
+const createFunc = jest.fn()
+
+//TODO: Put in test data
+
+const getTestFile = (abs: AbsolutePath, cloudProvider: CloudProviderType = cloudProviderTypes.GCP): AnalysisFile => ({
+  name: abs,
+  ext: '.ipynb' as Extension,
+  displayName: getDisplayName(abs),
+  fileName: getFileName(abs),
+  tool: getToolFromFileExtension(getExtension(abs)) as ToolLabel,
+  lastModified: new Date().getTime(),
+  cloudProvider
+})
+
+
+describe('AnalysisModal', () => {
+  beforeEach(() => {
+    // Arrange
+    asMockedFn(useAnalysisFiles).mockImplementation(() => ({
+      refresh: () => Promise.resolve(),
+      loadedState: { state: [], status: 'Ready' },
+      create: createFunc,
+      pendingCreate: { status: 'Ready', state: true }
+    }))
+
+    //@ts-expect-error
+    Ajax.mockImplementation(() => ({ Metrics: { captureEvent: jest.fn() } }))
+  })
+
+  it('GCP - Renders correctly by default', () => {
+    // Act
+    render(h(AnalysisModal, defaultGcpModalProps))
+    // Assert
+    screen.getByText('Select an application')
+    screen.getByAltText('Create new notebook')
+    screen.getByAltText('Create new R file')
+    screen.getByAltText('Create new Galaxy app')
+  })
+
+  it('GCP - Successfully resets view.', async () => {
+    // Act
+    render(h(AnalysisModal, defaultGcpModalProps))
+
+    // Act
+    const button = screen.getByAltText('Create new notebook')
+
+    await userEvent.click(button)
+    screen.getByText('Create a new notebook')
+
+    const backButton = screen.getByLabelText('Back')
+    await userEvent.click(backButton)
+
+    // Assert
+    screen.getByText('Select an application')
+  })
+
+  it.each([
+    { app: 'Jupyter', buttonAltText: 'Create new notebook', expectedTitle: 'Create a new notebook' },
+    { app: 'RStudio', buttonAltText: 'Create new R file', expectedTitle: 'Create a new R file' },
+    { app: 'Galaxy', buttonAltText: 'Create new Galaxy app', expectedTitle: 'Galaxy Cloud Environment' }
+  ])('GCP - Renders correctly and selects $app when no apps or runtimes are present.', async ({ buttonAltText, expectedTitle }) => {
+    // Arrange
+    render(h(AnalysisModal, defaultGcpModalProps))
+
+    // Act
+    const button = screen.getByAltText(buttonAltText)
+
+    await userEvent.click(button)
+
+    // Assert
+    screen.getByText(expectedTitle)
+  })
+
+  it.each([
+    { fileType: 'Python 3' },
+    { fileType: 'R' },
+  ])('GCP - Creates a new $fileType for Jupyter when no apps or runtimes are present and navigates to environment creation.', async ({ fileType }) => {
```

Comment:
nit: Maybe rephrase this slightly from `navigates to environment creation.` to `opens environment creaton modal`. I was looking for asserts around `nav.goToxxx` until I realized what this meant

Reply:
Good feedback!